packages/core/http/core-http-server-internal/src/lifecycle_handlers.test.ts

```diff
@@ -424,3 +439,31 @@ describe('customHeaders pre-response handler', () => {
     });
   });
 });
+
+describe('lifecycle handler registration', () => {
+  it('will not register client version checking if disabled via config', () => {
+    const registrarMock = {
+      registerAuth: jest.fn(),
+      registerOnPostAuth: jest.fn(),
+      registerOnPreAuth: jest.fn(),
```

Comment:
NIT: you extracted the registration to `register_lifecycle_handlers.ts`, so the associated tests should be in a dedicated `register_lifecycle_handlers.test.ts` file.